### ISSUE TITLE
[etc/prefer-interface]: Does not report constructor signature

### ISSUE THREAD

**Issue by ilyub** [cartant/eslint-plugin-etc#44]
```ts
type f = () => void;

type g = new () => void;
```

prefer-interface reports f, but does not report g.
